resources/Microsoft.Windows.Setting.Language/Microsoft.Windows.Setting.Language.psm1

```diff
@@ -0,0 +1,213 @@
+    $global:LocaleNameRegistryPath = 'HKCU:\Control Panel\International'
+    $global:LocaleUserProfilePath = 'HKCU:\Control Panel\International\User Profile'
+} else {
+    $global:LocaleNameRegistryPath = $global:LocaleUserProfilePath = $env:TestRegistryPath
+}
+
+#region Functions
+function Get-OsBuildVersion {
+    return [System.Environment]::OSVersion.Version.Build
+}
+
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )    
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key | Select-Object -ExpandProperty $Property)     
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+
+function Get-LocaleList {
+    $localeList = Get-WinUserLanguageList
+    $out = [List[Language]]::new()
+
+    foreach ($locale in $localeList) {
+        $language = [Language]::new($locale.LanguageTag, $true)
+        $out.Add($language)
+    }
+
+    # section to include other languages that can be installed
+    # helpful for users to discover what packages can be installed
+    $allLangues = [System.Globalization.CultureInfo]::GetCultures('AllCultures')
+    foreach ($culture in $allLangues) {
+        if ($out.LocaleName -notcontains $culture.Name -and -not ([string]::IsNullOrEmpty($culture.Name))) {
+            $language = [Language]::new($culture.Name, $false)
+            $out.Add($language)
+        }
+    }
+
+    return $out
+}
+#endregion Functions
+
+#region Classes
+<#
+.SYNOPSIS
+    The `Language` DSC Resource allows you to install, update, and uninstall languages on your local Windows machine.
+
+.PARAMETER LocaleName
+    The name of the language. This is the language tag that represents the language. For example, `en-US` represents English (United States).
+    To get a full list of languages available, use the `Get-LocaleList` function or Export() method.
+
+.PARAMETER Exist
+    Indicates whether the package should exist. Defaults to $true.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName Microsoft.Windows.Setting.Language -Name Language -Method Set -Property @{ LocaleName = 'en-US' }
+
+    This example installs the English (United States) language on the local machine.
+#>
+[DscResource()]
+class Language {
+    [DscProperty(Key)]
+    [string] $LocaleName
+
+    [DscProperty()]
+    [bool] $Exist = $true
+
+    static [hashtable] $InstalledLocality
+
+    Language() {
+        [Language]::GetInstalledLocality()
+    }
+
+    Language([string] $LocaleName, [bool] $Exist) {
+        $this.LocaleName = $LocaleName
+        $this.Exist = $Exist
+    }
+
+    [Language] Get() {
+        $keyExist = [Language]::InstalledLocality.ContainsKey(($this.LocaleName))
+
+        $currentState = [Language]::InstalledLocality[$this.LocaleName]
```

Comment:
You check if the key exists, then you immediately look up the value in the map. Seems like the key exists check needs to here.